dist/metacom.js

```diff
@@ -85,33 +117,45 @@ export class Metacom extends EventEmitter {
           return;
         }
         resolve(args);
-        return;
-      }
-      if (callType === 'event') {
+      } else if (callType === 'event') {
         const [interfaceName, eventName] = target.split('/');
         const metacomInterface = this.api[interfaceName];
         metacomInterface.emit(eventName, args);
-      }
-      if (callType === 'stream') {
-        const { name, size, status } = packet;
-        if (name) {
-          const stream = { name, size, chunks: [], received: 0 };
-          this.streams.set(callId, stream);
-          return;
-        }
-        const stream = this.streams.get(callId);
-        if (status) {
-          this.streams.delete(callId);
-          const blob = new Blob(stream.chunks);
-          blob.text().then((text) => {
-            console.log({ text });
-          });
-          return;
+      } else if (callType === 'stream') {
+        const { stream: streamId, name, size, status } = packet;
+        const stream = this.streams.get(streamId);
+        if (name && typeof name === 'string' && Number.isSafeInteger(size)) {
+          if (stream) {
+            console.error(new Error(`Stream ${name} is already initialized`));
+          } else {
+            const streamData = { streamId, name, size };
+            const stream = new MetacomReadable(streamData);
+            this.streams.set(streamId, stream);
+          }
+        } else if (!stream) {
+          console.error(new Error(`Stream ${streamId} is not initialized`));
+        } else if (status === 'end') {
+          await stream.close();
+          this.streams.delete(streamId);
+        } else if (status === 'terminate') {
+          await stream.terminate();
+          this.streams.delete(streamId);
+        } else {
+          console.error(new Error('Stream packet structure error'));
```

Comment:
Long function, I'd like to decompose this

Reply:
Ok, I'll think on it